src/App.jsx

```diff
@@ -1,248 +1,272 @@
 /* eslint-disable jsx-a11y/accessible-emoji */
-import React from 'react';
+import React, { useState } from 'react';
 import './App.scss';
-
-// import usersFromServer from './api/users';
-// import categoriesFromServer from './api/categories';
-// import productsFromServer from './api/products';
-//
-// const products = productsFromServer.map((product) => {
-//   const category = null; // find by product.categoryId
-//   const user = null; // find by category.ownerId
-//
-//   return null;
-// });
-
-export const App = () => (
-  <div className="section">
-    <div className="container">
-      <h1 className="title">Product Categories</h1>
-
-      <div className="block">
-        <nav className="panel">
-          <p className="panel-heading">Filters</p>
-
-          <p className="panel-tabs has-text-weight-bold">
-            <a
-              data-cy="FilterAllUsers"
-              href="#/"
-            >
-              All
-            </a>
-
-            <a
-              data-cy="FilterUser"
-              href="#/"
-            >
-              User 1
-            </a>
-
-            <a
-              data-cy="FilterUser"
-              href="#/"
-              className="is-active"
-            >
-              User 2
-            </a>
-
-            <a
-              data-cy="FilterUser"
-              href="#/"
-            >
-              User 3
-            </a>
-          </p>
-
-          <div className="panel-block">
-            <p className="control has-icons-left has-icons-right">
-              <input
-                data-cy="SearchField"
-                type="text"
-                className="input"
-                placeholder="Search"
-                value="qwe"
-              />
-
-              <span className="icon is-left">
-                <i className="fas fa-search" aria-hidden="true" />
-              </span>
-
-              <span className="icon is-right">
-                {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-                <button
-                  data-cy="ClearButton"
-                  type="button"
-                  className="delete"
-                />
-              </span>
+import cn from 'classnames';
+
+import usersFromServer from './api/users';
+import categoriesFromServer from './api/categories';
+import productsFromServer from './api/products';
+
+function getOwnerById(ownerId) {
+  return usersFromServer.find(user => user.id === ownerId) || null;
+}
+
+function getCategoryById(categoryId) {
+  return categoriesFromServer
+    .find(category => category.id === categoryId) || null;
+}
+
+export const combinedProducts = productsFromServer.map(product => ({
+  ...product,
+  user: getOwnerById(getCategoryById(product.categoryId).ownerId),
```

Comment:
Pay attention that `getCategoryById` can possibly return null

Reply:
Yes, totally agree with you. Again, rush-coding

Reply:
Yes, I totally agree with you. Again, rush-coding...

Reply:
Yes, I totally agree with you. Again, rush-coding...

Reply:
Yes, totally agree with you. Again, rush-coding